### PR TITLE
Docker 1.10 Improvements

### PR DESCRIPTION
**Pull request by rolandpoulter**
* Updated VM to run Docker 1.10
* Updated `docker-compose.yml` to version 2
  * Added `depends_on` to help start the services in order.
  * Added `build` to make it easier to build RackHD docker images. - #100 
* Separated isc-dhcp-server from `on-taskgraph` and created shared mounts for dhcpd.leases file. - #99 
* Created `rackhd/files` image that downloads pxe images from bintray, then shares the files with host, `on-http` and `on-tftp`.
  * Updated `on-http` and `on-tftp` Dockerfiles to remove file downloads.
* Fixes for `monorail/config.json`.
* Added `kibana.yml` config file.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -4,29 +4,20 @@ Copyright 2016, EMC, Inc.
...
 $ docker-compose up           # Run RackHD on Docker with ELK.
 ```
 
-Exposed Services:
-* http://127.0.0.1:15672/ - RabbitMQ Managment
-* http://127.0.0.1:5601/ - Kibana
-* http://127.0.0.1:9090/docs - RackHD Docs
-* http://127.0.0.1:9090/ui - RackHD UI
-* http://127.0.0.1:9200/ - Elasticsearch
+Then goto: http://127.0.0.1:9090
```
Comment (benbp) [inline docker/README.md]: "goto"

**Comment by benbp** [comment]
:+1:

File: docker/files/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+RUN mkdir -p /RackHD/downloads
+
+ADD https://bintray.com/artifact/download/rackhd/binary/ipxe/monorail-efi32-snponly.efi \
+    /RackHD/downloads/monorail-efi32-snponly.efi
+ADD https://bintray.com/artifact/download/rackhd/binary/ipxe/monorail-efi64-snponly.efi \
+    /RackHD/downloads/monorail-efi64-snponly.efi
+ADD https://bintray.com/artifact/download/rackhd/binary/ipxe/monorail.ipxe \
+    /RackHD/downloads/monorail.ipxe
+ADD https://bintray.com/artifact/download/rackhd/binary/ipxe/undionly.kpxe \
```
Comment (anhou) [inline docker/files/Dockerfile]: Remove 'ADD https://bintray.com/artifact/download/rackhd/binary/ipxe/undionly.kpxe' and add 'https://bintray.com/artifact/download/rackhd/binary/syslinux/undionly.kkpxe', it's the latest update in master branch

**Comment by anhou** [comment]
:+1: after address the comments

**Comment by anhou** [comment]
Jenkins failure is unrelated with docker, so merge it

**Pull request merged**